src/ahp/prover.rs

```diff
@@ -296,10 +296,16 @@ impl<F: PrimeField> AHPForR1CS<F> {
                 w
             }
         };
-        let (w_poly, remainder) = w_poly.divide_by_vanishing_poly(&domain_x).unwrap();
-        assert!(remainder.is_zero()); // w_poly is divisible by v_X because we set w = 0 on X.
+        let (w_poly, remainder) = match w_poly.divide_by_vanishing_poly(&domain_x) {
+            Some(v) => v,
```

Comment:
Let's keep the unwrap and assert here: as stated by the comment, this division mustn't fail and must return a 0 remainder. If it happens, it means there is a software bug